lib/socks.c

```diff
@@ -68,7 +70,9 @@ int Curl_blockread_all(struct connectdata *conn, /* connection data */
       result = CURLE_OPERATION_TIMEDOUT;
       break;
     }
-    if(SOCKET_READABLE(sockfd, timeleft) <= 0) {
+    if(sizeof(time_t) < 8 && timeleft > INT_MAX)
```

Comment:
We have a `SIZEOF_TIME_T` define that lets you do this check only on those platforms instead of checking size at run-time. And we have `TIME_T_MAX` too.

Reply:
Thanks, done.